sbi/neural_nets/net_builders/mnle.py

```diff
@@ -144,15 +151,31 @@ def build_mnle(
     combined_condition = torch.cat([disc_x, embedded_batch_y], dim=-1)
 
     # Set up a categorical RV neural net for modelling the discrete data.
-    discrete_net = build_categoricalmassestimator(
-        disc_x,
-        batch_y,
-        z_score_x="none",  # discrete data should not be z-scored.
-        z_score_y="none",  # y-embedding net already z-scores.
-        num_hidden=hidden_features,
-        num_layers=hidden_layers,
-        embedding_net=embedding_net,
-    )
+    if categorical_model == "made":
+        discrete_net = build_autoregressive_categoricalmassestimator(
+            disc_x,
+            batch_y,
+            z_score_x="none",  # discrete data should not be z-scored.
+            z_score_y="none",  # y-embedding net already z-scores.
+            num_hidden=hidden_features,
+            num_layers=hidden_layers,
+            embedding_net=embedding_net,
+        )
+    elif categorical_model == "mlp":
+        assert num_disc == 1, "MLP only supports 1D input."
+        discrete_net = build_categoricalmassestimator(
+            disc_x,
+            batch_y,
+            z_score_x="none",  # discrete data should not be z-scored.
+            z_score_y="none",  # y-embedding net already z-scores.
+            num_hidden=hidden_features,
+            num_layers=hidden_layers,
+            embedding_net=embedding_net,
+        )
```

Comment:
more generally, isn't the MLP a special case of the MADE? can't we absorb them into one class?

Reply:
Comment: Check if testcase is identical, if yes -> rm MLP